verifcid/validate.go

```diff
@@ -3,57 +3,65 @@ package verifcid
 import (
 	"fmt"
 
-	cid "github.com/ipfs/go-cid"
+	"github.com/ipfs/go-cid"
 	mh "github.com/multiformats/go-multihash"
 )
 
 var ErrPossiblyInsecureHashFunction = fmt.Errorf("potentially insecure hash functions not allowed")
 var ErrBelowMinimumHashLength = fmt.Errorf("hashes must be at least %d bytes long", minimumHashLength)
 var ErrAboveMaximumHashLength = fmt.Errorf("hashes must be at most %d bytes long", maximumHashLength)
 
-const minimumHashLength = 20
-const maximumHashLength = 128
+// Allowlist defines an interface containing list of allowed multihashes.
+type Allowlist interface {
+	// IsAllowed checks for multihash allowance by the code.
+	IsAllowed(code uint64) bool
+}
 
-var goodset = map[uint64]bool{
-	mh.SHA2_256:     true,
-	mh.SHA2_512:     true,
-	mh.SHA3_224:     true,
-	mh.SHA3_256:     true,
-	mh.SHA3_384:     true,
-	mh.SHA3_512:     true,
-	mh.SHAKE_256:    true,
-	mh.DBL_SHA2_256: true,
-	mh.KECCAK_224:   true,
-	mh.KECCAK_256:   true,
-	mh.KECCAK_384:   true,
-	mh.KECCAK_512:   true,
-	mh.BLAKE3:       true,
-	mh.IDENTITY:     true,
-
-	mh.SHA1: true, // not really secure but still useful
+// NewAllowlist constructs new [Allowlist] from the given map set.
+func NewAllowlist(allowset map[uint64]bool) Allowlist {
+	return allowlist{allowset: allowset}
 }
 
-func IsGoodHash(code uint64) bool {
-	good, found := goodset[code]
-	if good {
-		return true
+// NewOverdingAllowlist is like [NewAllowlist] but it will fallback to an other [AllowList] if keys are missing.
+// If override is nil it will return unsecure for unknown things.
+func NewOverdingAllowlist(override Allowlist, allowset map[uint64]bool) Allowlist {
+	return allowlist{override, allowset}
+}
+
+type allowlist struct {
+	override Allowlist
+	allowset map[uint64]bool
+}
+
+func (ghr allowlist) IsAllowed(code uint64) bool {
+	if good, found := ghr.allowset[code]; found {
+		return good
 	}
 
-	if !found {
-		if code >= mh.BLAKE2B_MIN+19 && code <= mh.BLAKE2B_MAX {
-			return true
-		}
-		if code >= mh.BLAKE2S_MIN+19 && code <= mh.BLAKE2S_MAX {
-			return true
-		}
+	if ghr.override != nil {
+		return ghr.override.IsAllowed(code)
 	}
 
 	return false
 }
 
+const minimumHashLength = 20
+const maximumHashLength = 128
+
+// IsGoodHash checks for multihash allowance by the given code.
+func IsGoodHash(code uint64) bool {
+	return DefaultAllowlist.IsAllowed(code)
+}
+
+// ValidateCid validates multihash allowance behind given CID.
 func ValidateCid(c cid.Cid) error {
+	return ValidateCidWithAllowlist(DefaultAllowlist, c)
+}
+
+func ValidateCidWithAllowlist(allowlist Allowlist, c cid.Cid) error {
```

Comment:
```suggestion
func ValidateCid(allowlist Allowlist, c cid.Cid) error {
```

There is a very little amount of places which uses this code, removing the defaults querier will force all future consumers to use the correct dependency.

Reply:
Let's also then change the name of the pkg. verifcid should become verifmh or something